05-Mining-Protocol.md

```diff
@@ -448,3 +518,57 @@ This message can be sent only to connections that don’t have `REQUIRES_STANDAR
 | ---------------- | ------------- | ----------------------------------------------------------------------------------------- |
 | group_channel_id | U32           | Identifier of the group where the standard channel belongs                                |
 | channel_ids      | SEQ0_64K[U32] | A sequence of opened standard channel IDs, for which the group channel is being redefined |
+
+## 5.5 Proxies
+
+Proxies are entities that aggregate hashrate from multiple Mining Devices into a single Connection.
+
+The following protocol primitives are based on the assumption that proxies are part of the infrastructure:
+- Extended Extranonces: allowing for efficient hashing space distribution.
+- Future Jobs: allowing for in-site caching of Future Jobs (therefore avoiding empty Jobs).
+
+Proxies can be nested into multiple layers, depending on different needs for hasher infrastructure.
+The infrastructure complexity may vary according to operational requirements, but a single proxy layer should suffice for most simple setups.
+
+Proxies can be split in two main categories:
+
+### 5.5.1 Mining Proxy
+
+A Mining Proxy covers the full scope of the Mining Protocol. It supports all types of channels, both for Upstream and Downstream Connections.
+
+|                | **Standard Channels** | **Group Channels** | **Extended Channels** |
+|----------------|-----------------------|--------------------|-----------------------|
+| **Upstream**   | yes                   | yes                | yes                   |
+| **Downstream** | yes                   | yes                | yes                   |
+
+A Mining Proxy is responsible for delivering Jobs to Downstream Channels and is a crucial piece of infrastructure for optimizing the hash space distribution across multiple SV2 Mining Devices (plus SV1 Mining Devices, in case a Translator Proxy is available Downstream).
+
+A Mining Proxy can forward multiple Standard Channels to the Upstream, ideally one representing each SV2 Mining Device Downstream.
+The Upstream (Pool, JDC or another Mining Proxy) is expected to send a `SetGroupChannel` message aggregating different Standard Channels.
+The Upstream sends jobs via `NewExtendedMiningJob` to Group Channels, and if some Downstream is a Mining Device (i.e.: the `SetupConnection` had the `REQUIRES_STANDARD_JOBS` flag) the Mining Proxy converts that message into different `NewMiningJob` messages after calculating the correct Merkle Root based on each Standard Channel's `extranonce_prefix`.
+
+Alternatively, a Mining Proxy could aggregate multiple Donwstream Standard Channels into a single Extended Channel. The Upstream sends Jobs via `NewExtendedMiningJob`, and for each Downstream SV2 Mining Device, the Mining Proxy sends different `NewMiningJob` message, where the Merkle Root is based on the Standard Channel's `extranonce_prefix`.
```

Comment:
What could be the reason why a proxy prefers to open many standard channels (with upstream), which are then grouped in a group channel? Why that proxy is not directly opening an extended channel with upstream?

Reply:
I asked a similar question when I was looking at implementation details

https://github.com/stratum-mining/stratum/discussions/1145#discussioncomment-10465172